gateway/tests/api/test_files.py

```diff
@@ -0,0 +1,80 @@
+"""Tests files api."""
+import os
+from urllib.parse import quote_plus
+
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+
+class TestFilesApi(APITestCase):
+    """TestProgramApi."""
+
+    fixtures = ["tests/fixtures/fixtures.json"]
+
+    def test_files_list_non_authorized(self):
+        """Tests files list non-authorized."""
+        url = reverse("v1:files-list")
+        response = self.client.get(url, format="json")
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_files_list(self):
+        """Tests files list."""
+
+        media_root = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            "..",
+            "resources",
+            "fake_media",
+        )
+
+        with self.settings(MEDIA_ROOT=media_root):
+            auth = reverse("rest_login")
+            response = self.client.post(
+                auth, {"username": "test_user", "password": "123"}, format="json"
+            )
+            token = response.data.get("access")
+            self.client.credentials(HTTP_AUTHORIZATION="Bearer " + token)
+            url = reverse("v1:files-list")
+            response = self.client.get(url, format="json")
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            self.assertEqual(response.data, {"results": ["artifact.tar"]})
+
+    def test_non_existing_file_download(self):
```

Comment:
not sure how much value we'd actually get from this , but it might be good to add a test trying to escape the user's directory (more so that if it ever does become possible, we get an alert). a decent chance that it doesn't tell us anything we don't already know though...

Reply:
Good suggestion! Added in [abf2247](https://github.com/Qiskit-Extensions/quantum-serverless/pull/896/commits/abf22471362ec7d559ce32fe816c47616672194f)